src/docfx/build/toc/BuildTableOfContents.cs

```diff
@@ -20,8 +21,9 @@ public static List<Error> Build(Context context, Document file)
 
             if (file.Docset.Config.Output.Pdf)
             {
-                model.Metadata.PdfAbsolutePath = "/" + UrlUtility.Combine(
-                    file.Docset.SiteBasePath, "opbuildpdf", LegacyUtility.ChangeExtension(file.SitePath, ".pdf"));
+                var siteBasePath = file.Docset.SiteBasePath;
+                var relativePath = Path.GetRelativePath(string.IsNullOrEmpty(siteBasePath) ? "." : siteBasePath, LegacyUtility.ChangeExtension(outputPath, ".pdf"));
+                model.Metadata.PdfAbsolutePath = PathUtility.NormalizeFile(Path.Combine("/", siteBasePath, "opbuildpdf", relativePath));
```

Comment:
Will this work? I was trying to avoid doing reverse operation (`GetRelativePath`)
```suggestion
                model.Metadata.PdfAbsolutePath = PathUtility.NormalizeFile(Path.Combine("/", siteBasePath, "opbuildpdf", monikerHash, relativePath));
```

Reply:
+1.
It could just be 
```csharp
model.Metadata.PdfAbsolutePath = "/" + UrlUtility.Combine(
                    file.Docset.SiteBasePath, "opbuildpdf", MonikerUtility.GetGroup(model.Metadata.Monikers) ?? "", LegacyUtility.ChangeExtension(file.SitePath, ".pdf"));
```

Reply:
got your point